staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -193,13 +193,13 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		if err != nil {
 			return nil, err
 		}
-		mergo.MergeWithOverwrite(clientConfig, userAuthPartialConfig)
```

Comment:
MergeWithOverwrite is deprecated?

Reply:
Yes https://github.com/imdario/mergo/blob/master/merge.go#L296

Reply:
we seem to be ping-ponging back and forth on this library, which is unfortunate - https://github.com/kubernetes/kubernetes/pull/65779

Reply:
@ash2k could you track why we didn't go your route in #65779 ? As far as I see the override option already existed back then, added in 2017 or so.

Reply:
@sttts You mean why a deprecated function was introduced? I don't know and it wasn't discussed. The change in this PR does not change what is happening under the hood, as the current codepath just delegates to the new codepath - see https://github.com/imdario/mergo/blob/v0.3.5/merge.go#L194-L219

Reply:
no, why did we switch to a deprecated function in the PR @liggitt referenced. If it has been just an oversight, his argument of ping-pong goes away.

Reply:
@sttts In #65779 we switched from `Merge()` to `MergeOverride()` for some reason. It was a change in behavior. Here I'm not changing the current behavior, I'm just replacing `MergeOverride()` with `Merge(WithOverride())` which are 100% identical (see the link above).

I think it was just an oversight that a deprecated function was used i.e. there is no ping-pong :)

Reply:
:+1:

Reply:
So I agree we should move away from the deprecated func and move forward.